libs/runtime/kernel/ebpf_platform_kernel.c

```diff
@@ -361,6 +362,83 @@ _IRQL_requires_max_(PASSIVE_LEVEL) _Must_inspect_result_ ebpf_result_t
     return EBPF_SUCCESS;
 }
 
+// Function to convert UTF-8 string to Unicode string.
+static ebpf_result_t
+_convert_utf8_string_to_unicode_string(
+    _Out_ PUNICODE_STRING destination_string, _In_ PCSZ source_string, _In_ BOOLEAN allocate_destination_string)
+{
+    EBPF_LOG_ENTRY();
+    ebpf_result_t result = EBPF_SUCCESS;
+    NTSTATUS status;
+    ULONG bytes_in_unicode_string = 0;
+
+    if (!destination_string || !source_string) {
+        result = EBPF_INVALID_ARGUMENT;
+        goto Done;
+    }
+
+    destination_string->Buffer = NULL;
+    destination_string->Length = 0;
+    destination_string->MaximumLength = 0;
+
+    ULONG source_length = (ULONG)strlen(source_string);
+
+    // First pass: Calculate how many bytes are needed for the Unicode string.
```

Comment:
```suggestion
    // First pass: Calculate how many bytes are needed for the UTF-16LE string.
```
or "... for the UNICODE_STRING" would also be ok, but it's less human understandable.